src/NHibernate/Multi/QueryBatch.cs

```diff
@@ -44,18 +47,17 @@ public void Execute()
 			{
 				Init();
 
-				if (!Session.Factory.ConnectionProvider.Driver.SupportsMultipleQueries)
+				using (Session.BeginProcess())
 				{
-					foreach (var query in _queries)
+					if (!Session.Factory.ConnectionProvider.Driver.SupportsMultipleQueries)
 					{
-						query.ExecuteNonBatched();
+						foreach (var query in _queries)
+						{
+							query.ExecuteNonBatched();
+						}
+						return;
 					}
-					return;
-				}
-
-				using (Session.BeginProcess())
-				{
-					DoExecute();
+					ExecuteBatched();
```

Comment:
Unrelated slight improvement here: renamed for consistency with `ExecuteNonBatched`.